components/courses/AssessmentsTimeline.vue

```diff
@@ -0,0 +1,174 @@
+      return [...this.assessments].sort((a, b) => (a.start < b.start ? 1 : -1))
+    },
+    startdates() {
+      return this.sortedAssessments.map((assessment) =>
+        assessment.start ? assessment.start : this.minDate
+      )
+    },
+    enddates() {
+      return this.sortedAssessments.map((assessment) =>
+        assessment.end ? assessment.end : this.maxDate
+      )
+    },
+    assessementNames() {
+      return this.sortedAssessments.map((assessment) =>
+        this.shortName(assessment)
+      )
+    },
+    option() {
+      return {
+        tooltip: {
+          trigger: 'axis',
+          axisPointer: {
+            type: 'shadow',
+          },
+          formatter: (params) => {
+            const assessment = this.assessments.filter(
+              (assessment) => assessment.code === params[0].name
+            )[0]
+            let tooltip = assessment.name
+            if (assessment.start)
+              tooltip +=
+                '<br > ' +
+                this.$t('assessment.start.date', {
+                  start: this.formatDateFull(assessment.start),
+                })
+            if (assessment.end)
+              tooltip +=
+                '<br > ' +
+                this.$t('assessment.deadline.date', {
+                  deadline: this.formatDateFull(assessment.end),
+                })
+            return tooltip
+          },
+        },
+        xAxis: {
+          type: 'time',
+          position: 'top',
+          // Prevents echarts to freeze if boundaries are not ordered
+          min: this.minDate < this.maxDate ? this.minDate : this.maxDate,
+          max: this.minDate > this.maxDate ? this.minDate : this.maxDate,
+        },
+        yAxis: {
+          type: 'category',
+          data: this.assessementNames,
+          axisTick: {
+            alignWithLabel: true,
+          },
+          triggerEvent: true,
+        },
+        series: [
+          {
+            type: 'bar',
+            stack: 'total',
+            itemStyle: {
+              borderColor: '#fff',
+            },
+            z: 1,
+            data: this.enddates,
+          },
+          {
+            type: 'bar',
+            stack: 'total',
+            itemStyle: {
+              color: '#fff',
+            },
+            z: 3,
+            data: this.startdates,
+          },
+        ],
+      }
+    },
+  },
+
+  methods: {
+    goToAssessment(params) {
+      if (params.componentType === 'yAxis' && params.value) {
+        const assessment = this.assessments.filter(
+          (assessment) => assessment.code === params.value
+        )[0]
+        this.$router.push({
+          name: 'manage-courses-code-assessments-id',
+          params: {
+            code: this.code,
+            id: assessment.id,
+          },
+        })
+      }
+    },
+    shortName(assessment) {
+      if (assessment.code) return assessment.code
+      if (assessment.name.length < 10) return assessment.name
```

Comment:
Can be removed since from the specifications of slice, one can read that:
_If end is greater than the length of the sequence, slice extracts through to the end of the sequence (arr.length)._